### DISCUSSION TITLE
DMAC / OceanModeling coordination / Collaboration

### DISCUSSION THREAD

**Discussion by saeed-moghimi-noaa** [oceanmodeling/Management#18] [General]
DMAC / OceanModeling coordination / Collaboration

**Comment by dpsnowden** [comment]
Thanks @saeed-moghimi-noaa for getting this discussion started.   My hope for this conversation is twofold.  First, I hope that we can bring two communities together that are working on very similar problems.  [IOOS/DMAC](github.com/ioos) is a group of data managers working on tools to make it easier to manage, quality control, publish, and work with data from ocean observing systems (including ocean model products).   This group (e.g. @oceanmodeling/observational-data-retrieval and @oceanmodeling/model-wrapper etc) is focusing on the software tools needed to build and run the models.  There is a lot of overlap in our software needs and I'm hoping we can work together to identify promising technologies or software tools that we can work on together rather than duplicate efforts.  Second, we (the US IOOS Office in NOAA) host an [annual workshop](https://ioos.noaa.gov/project/dmac/) of data managers and software developers.  We have a meeting planned for 26-28 Sept 2023.  I'm hoping there are willing participants in this group to help create a session where we can share updates on recent software development and/or other common challenges.  Topics of interest could include (but aren't limited to):

* Understanding common workflows (for running or analyzing models or publishing model data)
* Understanding the needs for observational data for data assimilation, model verification/validation etc
* Latest tools to develop and run models in the cloud 
* Thoughts on software and/or tools that would enhance collaboration between NOAA modelers and those in Academia
* Latest strategies for publishing and/or analyzing large model results in the cloud (e.g. Pangeo for NOAA???) 

I'm sure the list could go on forever, hopefully this is a first step to building some excitement for a collaboration.  @mwengren and @mathewbiddle are the leads for the IOOS planning process.

**Reply by MathewBiddle** [reply]
More information about this year's IOOS DMAC Meeting can be found at https://groups.google.com/g/ioos_tech/c/sNOtw5g6v_8/m/KpttNBMRAAAJ

**Comment by saeed-moghimi-noaa** [comment]
Hi  @mwengren and @MathewBiddle

I will send an invite for you to join the oceanmodeling org.

Best,
@saeed-moghimi-noaa

**Comment by saeed-moghimi-noaa** [comment]
All,

Please see below the DMAC tentative agenda and some docs for a couple of breakout sessions which look very interesting.

I hope to see some of you join this event and some of the relevant breakouts. I hope we could address some questions like:

- How would IOOS DMAC be able to support the coastal ocean modeling community?
- Unification of observational data accessing tools (1D, 2D and 3D data). Also in relation to Data Assimilation for coastal ocean modeling.
- Tools to access structured and unstructured model generated 2D and 3D data.
- Tools to find and publish model generated data
- Tools to support users to subset and interpolate data on prem and on the cloud
- Open source development of tools with in [oceanmodeling ](https://github.com/oceanmodeling)org and beyond.

See here:
[Registration from](https://docs.google.com/forms/d/e/1FAIpQLScCz7nneQ7sBxfyskLYXA5H0UWS-NdFBMAMJV7dpWKwc0yyxg/viewform)



[2023 DRAFT DMAC Agenda](https://docs.google.com/document/d/1dbMR8DQxwTuhQzsLCaMxsoiPmEhDN5HYsRpgRRSq2Kk/edit#heading=h.bnf6iqipnu9x)
[2023 IOOS DMAC Agenda Extra info](https://docs.google.com/document/d/10fuzZM8UTXmxtpcdK075Q41BKR002r-7w9vInlKpvjA/edit#heading=h.1rjnylufnqw4)
[DMAC 2023 Breakout: Modeling use cases](https://docs.google.com/document/d/1O9BpAOwpII1WaFuJauZw51c_lArtVIa8sogfIjjOR8o/edit?usp=sharing)
[DMAC 2023 Breakout: DOI generation by NCEI](https://docs.google.com/document/d/1T5kGFVXfTh4423S7p9p8A6qtExfQacBi5VqWwtlWx9c/edit?usp=sharing)

Best,
-Saeed

**Comment by saeed-moghimi-noaa** [comment]
more info

**Comment by dpsnowden** [comment]
Updating this with information about the 2024 DMAC meeting.

We are pleased to announce that the 2024 DMAC Meeting will take place virtually on Tuesday April 2 to Thursday April 4, 2024 in the afternoons between 1 pm and 5 pm ET.



Please save these dates in your calendars.  



Agenda Development:



To help us identify topics of interest to the IOOS DMAC community, we've created an agenda input document (linked below).  Please contribute your ideas for both plenary session presentations as well as for breakout discussion and/or workshop sessions.  Please complete all fields, including your name as topic suggester, so that we may follow up if needed.  



2024 DMAC Annual Meeting Agenda Input:

[Draft Agenda](https://docs.google.com/document/d/1dbMR8DQxwTuhQzsLCaMxsoiPmEhDN5HYsRpgRRSq2Kk/edit)


Please contribute your ideas by Friday March 1.

For any questions pertaining to the agenda or the meeting in general, email the IOOS DMAC Team at [data.ioos@noaa.gov](mailto:data.ioos@noaa.gov).

**Reply by saeed-moghimi-noaa** [reply]
Thanks @dpsnowden

**Reply by MathewBiddle** [reply]
FYI, the agenda input document is now closed. We are working on building a draft Agenda to be shared soon.

**Comment by saeed-moghimi-noaa** [comment]
All, 

As it was decided in DMAC breakout meeting today.
![image](https://github.com/oceanmodeling/Management/assets/66274703/d116018d-cc63-4192-8099-37fce181984f)


There will be a follow up conversation on: **Requirements Document”: outline of what features / requirements people want. This could guide folks developing individual pieces of the puzzle.**

Please tag or add a note using your github account so we can organize meetings / follow ups.

Thanks,

**Reply by saeed-moghimi-noaa** [reply]
[Breakout Note: ](https://docs.google.com/document/d/1W1MV9yXpw22d9f5Ui4cgYXAn05v89InlFDG6AaqgEbY/edit)

**Reply by Armaghan-NOAA** [reply]
a note from me!

**Reply by ChrisBarker-NOAA** [reply]
Getting myself (Chris Barker) officially in the conversation with my gitHub Handle

**Reply by tshyka** [reply]
I'd like to join the conversation too.

**Reply by cflindley-noaa** [reply]
same same

**Comment by ChrisBarker-NOAA** [comment]
Develop a Topic for the 2024 IOOS Sprint May21-13:

https://ioos.github.io/ioos-code-sprint/2024/

It would be great to have a a definition for what this project / topic is -- then we can either use the Sprint to develop requirements docs, etc, or, if we've gotten a bit farther, actually work on code -- or at least testing, etc.

My first draft of a title:

### Developing unified code bases / APIs for accessing and working with oceanographic model results.
